VitruvioHost/Plugins/Vitruvio/Source/Vitruvio/Private/VitruvioBatchActor.cpp

```diff
@@ -394,12 +417,15 @@ void AVitruvioBatchActor::ProcessGenerateQueue()
 
 		ProcessGenerateQueueCriticalSection.Unlock();
 
-		for (int ComponentIndex = 0; ComponentIndex < Item.VitruvioComponents.Num(); ++ComponentIndex)
+		if (Item.GenerateResultDescription.EvaluatedAttributes.Num() ==  Item.VitruvioComponents.Num())
```

Comment:
Maybe consider adding some comments for ifs where its not clear why they are necessary.